File: POO/Heritage/todo-sequence2/dame.cpp
#include "dame.h"

// Constructeur
Dame::Dame(const string nom/*=""*/, const string boissonFavorite/*="lait"*/, const string couleurRobe/*="blanche"*/) : Humain(nom, boissonFavorite), couleurRobe(couleurRobe), etat("libre")
{
}

// Accesseurs
string Dame::getNom() const
{
   return "Miss " + nom;
}

string Dame::getEtat() const
{
   return etat;
}

// Services
void Dame::sePresente() const
{
   cout << "(" << nom << ") -- " << "Bonjour, je suis " << this->getNom() << " et j'ai une jolie robe " << couleurRobe << endl;
}

void Dame::changeDeRobe(const string couleurRobe)
{
   this->couleurRobe = couleurRobe;
   cout << "(" << nom << ") -- " << "Regardez ma nouvelle robe " << couleurRobe << " !" << endl;
}

File: POO/Java/sequence5/src/western/Dame.java
package western;

public class Dame extends Humain
{
    private String boissonFavorite = "lait";
    private String couleurRobe = "blanche";
    private String etat = "";

    public Dame(String nom)
    {
        super(nom);
        setBoissonFavorite(boissonFavorite);
    }

    public void sePresente()
    {
        String texte = "";
        texte = "Bonjour, je suis Miss " + this.getNom() + " et j’ai une jolie robe " + this.getCouleurRobe();
        parle(texte);
    }

    public final String getCouleurRobe()
    {
        return couleurRobe;
    }

    public void changeDeRobe(String nouvelleCouleur)
    {
        String texte = "";
        couleurRobe = nouvelleCouleur;
        texte = "Regardez ma nouvelle robe " + couleurRobe + " !";
        parle(texte);
    }
}